test/master-playlist-controller.test.js

```diff
@@ -3111,5 +3110,87 @@ function(assert) {
                                 '#EXTINF:5.0\n' +
                                 '0.ts\n');
 
-  assert.equal(mpc.mediaSource.duration, Infinity, 'duration set on media source');
+  assert.equal(mpc.duration(), Infinity, 'duration reported as infinite');
+});
+
+QUnit.test('live playlist sets duration of media source to seekable end',
+function(assert) {
+  openMediaSource(this.player, this.clock);
+  const mpc = this.masterPlaylistController;
+
+  // master
+  this.standardXHRResponse(this.requests.shift());
+
+  assert.notOk(mpc.mediaSource.duration, 'no duration set on media source');
+
+  // since the safe live end will be 3 target durations back, in order for there to be a
+  // positive seekable end, there should be at least 4 segments
+  this.requests.shift().respond(200, null, `
+		#EXTM3U
+		#EXT-X-TARGETDURATION:5
+		#EXTINF:5
+		0.ts
+		#EXTINF:5
+		1.ts
+		#EXTINF:5
+		2.ts
+		#EXTINF:5
+		3.ts
+  `);
+
+  assert.equal(mpc.seekable().end(0), 5, 'calculated seekable end');
+  assert.equal(
+    mpc.mediaSource.duration,
+    mpc.seekable().end(0),
```

Comment:
you probably want to use real numbers to avoid the tests being tied to how the code is written

Reply:
Logically I wanted the tests to be that the duration matches with seekable end, but tested seekable end in the assertion above to verify that it is what we expect it to be. Do you think it would still be better to use the value on its own?

Reply:
ah, I see what you mean 🤔 I would just use the actual value to be safe and consistent